coralproject/talk
# [CORL-3120]: Add initialStatus to comment and use for pre-mod labels in moderate cards

## What does this PR do?

These changes add an `initialStatus` field to `Comment`. This helps have a clearer history from the initial creation of the comment into what its status history is. A `status` field is added to comment revisions, so we also have status per revision now. Then we use the first revision's status to populate `initialStatus`.

## These changes will impact:


## What changes to the GraphQL/Database Schema does this PR introduce?

This adds an `initialStatus` field to the `Comment`. It also adds `status` to `CommentRevision`.

## Does this PR introduce any new environment variables or feature flags?

no

## If any indexes were added, were they added to `INDEXES.md`?

n/a

## How do I test this PR?

You can set a user to pre-mod. Create comments as that user. See that they have the pre-mod label in moderate cards, in moderation queues and in user history drawer. Approve/reject these comments. See that they still have the pre-mod label based on their initial pre-mod status.

## Were any tests migrated to React Testing Library?



## How do we deploy this PR?
